climada/engine/impact.py

```diff
@@ -390,6 +392,24 @@ def impact_per_year(self, all_years=True, year_range=None):
             year_set[year] = sum(self.at_event[orig_year == year])
         return year_set
 
+    def impact_per_exp_reg(self, exposures):
+        """Aggregate impact matrix at the regional level, based on the
+        regions specified in the exposure.
+
+        Parameters
+        ----------
+        exposures : climada.entity.Exposures
+            exposure used to compute imp_mat
+        Returns
+        -------
+        np.matrix
+        """
+        self.at_reg_id_event = np.hstack([
+                       self.imp_mat[:, np.where(exposures.gdf.region_id == reg_id)[0]].sum(1)
```

Comment:
Does this work if `imp_mat` is an empty matrix? Does it make sense to then return zeros here or should this method throw an error instead?

Reply:
I am now returning `None` and throwing a warning when `imp_mat` is empty